src/libcore/alloc.rs

```diff
@@ -99,7 +99,7 @@ impl Layout {
     /// [`Layout::from_size_align`](#method.from_size_align).
     #[stable(feature = "alloc_layout", since = "1.28.0")]
     #[inline]
-    pub unsafe fn from_size_align_unchecked(size: usize, align: usize) -> Self {
+    pub const unsafe fn from_size_align_unchecked(size: usize, align: usize) -> Self {
```

Comment:
Doesn't it need `rustc_const_unstable` first?

Reply:
Otherwise, it's going to stabilize `const`ness immediately without any feature gate.

Reply:
This seems like a trivial addition that T-libs can just sign off on.

Reply:
@rasendubi, [core::num::NonZeroUsize::new_unchecked](https://doc.rust-lang.org/core/num/struct.NonZeroUsize.html#method.new_unchecked) is marked the same way.

Reply:
I believe the reason is that for `NonZeroUsize::new_unchecked` the `const`ness was introduced while it was unstable (https://github.com/rust-lang/rust/pull/48265/fileas#diff-01076f91a26400b2db49663d787c2576R95).

I'm not 100% sure though, so it would be nice if someone could confirm that.

Reply:
Would adding `#[rustc_const_unstable(feature="const_fn")]` resolve this? Documentation on the `rustc_const_unstable` macro seems to be sparse.

Reply:
If you go with `rustc_const_unstable`, then the constness will stay unstable and only be usable from nightly. We could then merge this PR without T-libs signoff. If you want this on stable, we're going to need that signoff anyway and I believe this function is simple enough and has clearly documented safety rules, so we can just stabilize it directly.